.circleci/config.yml

```diff
@@ -109,6 +109,46 @@ jobs:
             cargo install --force cargo-audit
             cargo audit
       - build_teardown
+  code-coverage:
+    description: Run code coverage
+    machine:
+      image: ubuntu-1604:201903-01
+    resource_class: large
+    steps:
+      - checkout
+      - run:
+          name: Install deps
+          command: |
+            sudo apt-get update
+            sudo apt-get install -y protobuf-compiler cmake curl clang
+            curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain none
+      - run:
+          name: Setup PATH
+          command: echo "export PATH=~/.cargo/bin:$PATH" >> $BASH_ENV
+      - env_setup
+      - run:
+          name: Install nightly toolchain (profile feature is not yet in beta/stable)
+          command: rustup install nightly
+      - run:
+          name: Install grcov
+          command: cargo install --force grcov
+      - run:
+          name: Setup rustc flags
+          command: |
+            echo "export CARGO_INCREMENTAL=0" >> $BASH_ENV
+            echo "export RUSTFLAGS='-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zno-landing-pads'" >> $BASH_ENV
+      - run:
+          name: Cargo test
+          command: cargo +nightly test
```

Comment:
I haven't managed to override the root `rust-toolchain` file's contents like that. The observable is that beta is picked, with the following error message:
```
Caused by:
  process didn't exit successfully: `sccache rustc - --crate-name ___ --print=file-names -Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Coverflow-checks=off -Zno-landing-pads -Cpasses=insert-gcov-profiling --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit code: 1)
--- stderr
error: the option `Z` is only accepted on the nightly compiler
```

Reply:
I got the hint from here https://github.com/rust-lang/rustup.rs#working-with-distribution-rust-packages